### ISSUE TITLE
[Excel] Addin is unusable in Chrome in a workbook with certain number of custom functions

### ISSUE THREAD

**Issue by sshukurov** [OfficeDev/office-js#2032]
## Context
Background to the issue is described in https://github.com/OfficeDev/office-js/issues/2031. I was able to reach a minimal repro where Excel Custom Functions addin is absolutely unusable in Chrome. I created a workbook with 8 sheets, each sheet containing 10K custom functions.

## Environment
* Platform: Office on the web
* Host: Excel
* Office version number: 16.0.14401.35900
* Operating System: Microsoft Windows 10 Pro 10.0.19042 Build 19042
* Browser: Google Chrome Version 92.0.4515.131 (Official Build) (64-bit)
 
## Expected behavior
Addin and custom functions are expected to work, though performance might reasonably degrade as the workbook size / the number of custom functions increase.

## Current behavior
The addin is absolutely unusable (custom functions are stuck, taskpane doesn't load) with `Can't Contact The Server` dialog shown upon loading the taskpane:
![image](https://user-images.githubusercontent.com/36232686/128551233-3fbd7e81-5db2-4d2b-aaa7-e95e49124f15.png)

## Steps to reproduce
1. Create a sample addin (Excel Custom Functions) [as described here](https://docs.microsoft.com/en-us/office/dev/add-ins/tutorials/excel-tutorial-create-custom-functions).

2. Build and sideload the addin to the workbook below (it contains 8 sheets with 10K formulas in each - the same `=CONTOSO.ADD(1, 2)` formula is repeated across a total of 80K cells: 
[WorkbookWithManyFunctions11.xlsx](https://github.com/OfficeDev/office-js/files/6947041/WorkbookWithManyFunctions11.xlsx)

3. Wait for the workbook to load, try to open the addin's taskpane.

## Useful logs
![image](https://user-images.githubusercontent.com/36232686/128551329-e1efd6b7-9399-4670-9b1b-b34eedc7d7d2.png)

**Comment by wh1t3cAt1k** [comment]
To give you guys some background, this minimal repro is based off a real workbook from one of our (Velixo) clients who has about 130000 custom functions in the workbook.

We got a support ticket complaining that the add-in was not refreshing data, and this is how we discovered the issue. 

The important point is that our legacy COM-based add-in built for Excel Desktop for Windows is perfectly usable at this size, and is rather fast to refresh. Unfortunately. the Office.js-based product, which we intended to market as the "better version, available across the platforms", it's still unusable due to this bug 😢
